src/Kestrel.Core/Internal/Http/HttpProtocol.FeatureCollection.cs

```diff
@@ -243,21 +243,14 @@ MinDataRate IHttpMinResponseDataRateFeature.MinDataRate
             get => MinResponseDataRate;
             set => MinResponseDataRate = value;
         }
-
-        object IFeatureCollection.this[Type key]
-        {
-            get => FastFeatureGet(key) ?? ConnectionFeatures[key];
-            set => FastFeatureSet(key, value);
-        }
-
-        TFeature IFeatureCollection.Get<TFeature>()
+        protected void ResetIHttpUpgradeFeature()
```

Comment:
Nit: missing newline

Reply:
Theoretically, isn't this exactly what `Set<IHttpUpgradeFeature>(this)` should optimize to?

Reply:
Comment below

Reply:
Added newline